### PR TITLE
Make adam variants more accessible

### PR DESCRIPTION
**Pull request by crcrpar**
This PR aims at making each Adam variant easier to initialize intuitively & more accessible.

### CODE REVIEW

**Comment by beam2d** [comment]
Sorry for the late response, I'll review it.

File: chainer/optimizers/adam.py [hunk 1]
```diff
@@ -426,3 +426,118 @@ def lr(self):
...
+        beta1 (float): Exponential decay rate of the first order moment.
+        beta2 (float): Exponential decay rate of the second order moment.
+        eta (float): Schedule multiplier, can be used for warm restarts.
+        weight_decay_rate (float): Weight decay rate.
+            The default value is 1.0.
+        eps (float): Small value for the numerical stability.
+    """
+
+    def __init__(self,
+                 alpha=_default_hyperparam,
```
Comment (beam2d) [inline chainer/optimizers/adam.py]:
```suggestion
                 alpha=_default_hyperparam.alpha,
```

File: chainer/optimizers/adam.py [hunk 2]
```diff
@@ -426,3 +426,118 @@ def lr(self):
...
+        eta (float): Schedule multiplier, can be used for warm restarts.
+    """
+
+    def __init__(self,
+                 alpha=_default_hyperparam.alpha,
+                 beta1=_default_hyperparam.beta1,
+                 beta2=_default_hyperparam.beta2,
+                 eps=_default_hyperparam.eps,
+                 eta=_default_hyperparam.eta):
+        super(Adam, self).__init__(
```
Comment (beam2d) [inline chainer/optimizers/adam.py]:
```suggestion
        super(AMSGrad, self).__init__(
```

File: chainer/optimizers/adam.py [hunk 3]
```diff
@@ -426,3 +426,118 @@ def lr(self):
...
+
+    def __init__(self,
+                 alpha=_default_hyperparam.alpha,
+                 beta1=_default_hyperparam.beta1,
+                 beta2=_default_hyperparam.beta2,
+                 final_lr=_default_hyperparam.final_lr,
+                 gamma=_default_hyperparam.gamma,
+                 eps=_default_hyperparam.eps,
+                 eta=_default_hyperparam.eta):
+        super(Adam, self).__init__(
```
Comment (beam2d) [inline chainer/optimizers/adam.py]:
```suggestion
        super(AdaBound, self).__init__(
```

File: chainer/optimizers/adam.py [hunk 4]
```diff
@@ -426,3 +426,118 @@ def lr(self):
...
```
Comment (beam2d) [inline chainer/optimizers/adam.py]:
```suggestion
        super(AMSBound, self).__init__(
```

**Comment by crcrpar** [comment]
I appreciate the comments.

**Comment by beam2d** [comment]
Jenkins, test this please.

**Comment by beam2d** [comment]
LGTM!

**Pull request merged**